core/src/main/java/org/apache/carbondata/core/metadata/datatype/DecimalConverterFactory.java

```diff
@@ -187,6 +197,43 @@ public void fillVector(Object valuesToBeConverted, int size,
             vector.putDecimal(i, value, precision);
           }
         }
+      } else if (pageType == DataTypes.BYTE_ARRAY) {
+        // complex primitive decimal dimension
+        int offset = 0;
+        for (int j = 0; j < size; j++) {
+          // here decimal data will be Length[4 byte], scale[1 byte], value[Length byte]
+          int len = ByteBuffer.wrap(data, offset, DataTypes.INT.getSizeInBytes()).getInt();
+          offset += DataTypes.INT.getSizeInBytes();
```

Comment:
DataTypes.INT.getSizeInBytes), It’s better to assign to some variable and use it inside loop, instead calling method every time

Reply:
done

Reply:
handled all the 5 files including base code